Add `exec` utility

ehmicky opened the pull request:
Fixes #218.

This adds an `exec` plugin utility for users to execute commands and processes.

This simply forwards to Execa as suggested by @DavidWells and @sw-yx in #218.

By using Execa, we also benefit from all its features (promise interface, cross-platform support, local binary execution, child process cleanup, interleaved output, better error messages, newline stripping, enhanced termination, etc.) and from the fact it's been battle-tested (currently used by [~4 million GitHub repositories](https://github.com/sindresorhus/execa/network/dependents?package_id=UGFja2FnZS0xODIxOTg0NQ%3D%3D)).

Example of usage (more examples are in the `README`):

```js
module.exports = {
  name: 'example-plugin',
  async onInit({ utils: { exec } }) {
    await exec('echo', ['hello', 'world'])
  }
}
```

The PR includes tests with 100% test coverage.

RaeesBhatti commented:
It seems like there are some symlinks in there, is that intended? @ehmicky

ehmicky commented:
Which ones?

DavidWells commented:
We need to iron out the final API before exposing this util to users.

Simplifying the API and adding a lightweight wrapper on this will protect us from being locked into a specific exec version

ehmicky commented:
Following our discussion on Slack, I have:
  - renamed `exec` to `run`
  - main interface: `run()` (arguments as array) vs `run.command()` (arguments as string)
  - run local binaries by default
  - print output to Netlify logs by default, but still allow users to retrieve the `stdout`/`stderr` as a return value
  
I've added new tests to cover those new cases.

We've also decided to only document (and support) the options from Execa that are strictly needed for most of our users. The `README` is still a WIP, so no options is documented yet (except the ones in the examples).
